src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml

```diff
@@ -102,5 +102,16 @@
             <argument type="service" id="translator"/>
             <argument type="string">%validator.translation_domain%</argument>
         </service>
+
+        <!-- LoginType -->
+        <service id="form.type.login" class="Symfony\Bundle\FrameworkBundle\Form\LoginType">
```

Comment:
Would it be possible to only make this form type available when there's a firewall that supports it? Might make the check in `buildForm` obsolete

Reply:
I don't think so. If someone wants to use it via a Guard authenticator, your proposal would disable the LoginType even though it could still be used elsewhere. IMO the LoginType should only depend on the request + a firewall, and maybe customizable options in the form itself.

Reply:
Currently it won't work with guard either way.

Reply:
I know, but I think it should